src/wallet/wallet2.cpp

```diff
@@ -5037,6 +5044,90 @@ uint64_t wallet2::adjust_mixin(uint64_t mixin)
   return mixin;
 }
 //----------------------------------------------------------------------------------------------------
+uint32_t wallet2::adjust_priority(uint32_t priority)
+{
+  if (priority == 0 && get_default_priority() != 1 && auto_low_priority())
+  {
+    try
+    {
+      // check if there's a backlog in the tx pool
+      const double fee_level = get_fee_multiplier(1) * get_per_kb_fee() / (double)1024;
```

Comment:
On second thought, this should use the "worst case" fee, since the actual tx may end up there.
The worst case is if your tx is an integer number of kB minus 1 byte, in which case it'll pay a comparatively low fee per byte compared to the average, which is what the above is checking.
So we need to check something like (fee_per_kB * 12 / 13) /1024, roughly.

Reply:
I'm not necessarily in full grasp of your point, but updated the code as suggested anyway. Does the number 12/13 come from the typical size of 2-in/2-out txes?

Reply:
Yes, typical tx is 13 kB.
With a x monero/kB fee, a 13 kB tx pays the same fee as a 14kB-1 byte one, but has a higher fee/byte ratio. So if you send the 14 kB-1 tx, it always gets pushed back to the end of the queue. The previous code was using the level for a 13 kB tx, so would not see the backlog caused by the 13 kB + some bytes ones, which the actual tx would likely be one of.